<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>talking_clock::SettingsDialog</class>
 <widget class="QDialog" name="talking_clock::SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>380</width>
    <height>280</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Talking Clock Settings</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="dialog_layout">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <!-- Hourly announcement group -->
   <item>
    <widget class="QGroupBox" name="hour_group">
     <property name="title">
      <string>Hourly</string>
     </property>
     <layout class="QGridLayout" name="hour_layout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="hour_sound_cb">
        <property name="text">
         <string>Sound</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="hour_sound_spacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <widget class="QToolButton" name="hour_sound_btn">
        <property name="icon">
         <iconset resource="../talking_clock.qrc">
          <normaloff>:/talking_clock/sound_icon.svg</normaloff>:/talking_clock/sound_icon.svg</iconset>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="hour_speech_cb">
        <property name="text">
         <string>Speech</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="hour_format_edit">
        <property name="placeholderText">
         <string notr="true">'北京时间，'AP h'点整'</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QToolButton" name="voice_config_btn">
        <property name="toolTip">
         <string>voice configuration</string>
        </property>
        <property name="icon">
         <iconset resource="../talking_clock.qrc">
          <normaloff>:/talking_clock/voice-config.svg</normaloff>:/talking_clock/voice-config.svg</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <!-- Quarter hour announcement group -->
   <item>
    <widget class="QGroupBox" name="quarter_group">
     <property name="title">
      <string>Quarter hour</string>
     </property>
     <layout class="QGridLayout" name="quarter_layout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="quarter_sound_cb">
        <property name="text">
         <string>Sound</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="quarter_sound_spacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <widget class="QToolButton" name="quarter_sound_btn">
        <property name="icon">
         <iconset resource="../talking_clock.qrc">
          <normaloff>:/talking_clock/sound_icon.svg</normaloff>:/talking_clock/sound_icon.svg</iconset>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="quarter_speech_cb">
        <property name="text">
         <string>Speech</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="2">
       <widget class="QLineEdit" name="quarter_format_edit">
        <property name="placeholderText">
         <string notr="true">'北京时间，'AP h'点'm'分'</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <!-- Quiet hours group -->
   <item>
    <widget class="QGroupBox" name="quiet_time_group">
     <property name="title">
      <string>Quiet time</string>
     </property>
     <layout class="QHBoxLayout" name="quiet_time_layout">
      <item>
       <widget class="QCheckBox" name="quiet_hours_enabled">
        <property name="text">
         <string>Enable</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTimeEdit" name="q_time_start"/>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string notr="true">-</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTimeEdit" name="q_time_end"/>
      </item>
     </layout>
    </widget>
   </item>
   <!-- Button box -->
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../talking_clock.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>talking_clock::SettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>talking_clock::SettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quiet_hours_enabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>q_time_start</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>154</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quiet_hours_enabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>q_time_end</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quiet_hours_enabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>label</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
